src/TS_HW3/Accounting.ts

```diff
@@ -0,0 +1,59 @@
+/*
+    Так само у нас є сутність Бухгалтерія, яка є департаментом і має властивість баланс, а також методи
+    для взяття на баланс співробітника або департаменту, зняття з балансу і виплати зарплати для всього персоналу.
+    Попередньо найняті співробітники отримують зарплату за допомогою зовнішніх оплат, Співробітники (тільки активні) - за допомогою внутрішніх.
+*/
+class Accounting extends Department {
+  public Balance: number;
+
+  constructor(company: Company, domain: string, debit: number, credit: number) {
+    super(company, DepartmentsNames.Accounting, domain, debit, credit);
+    this.Balance = debit - credit;
+  }
+
+  public TakeOnBalance(item: Employee | PreEmployee | Department): void {
+    switch (true) {
```

Comment:
Agreed, but I used switch-case just for varioty, because I've used if-else for type cheking in other method. Fistly 1st edition was for case1-case2-default, and then employee|preemployee  were joint to one case1.
Or did you mean thomething other regarding to opimality?

Reply:
No, I didn't. You got it right. I know what you mean, but constructions with literally true conditions may seem strange. :-)